packages/x-date-pickers-pro/src/models/fields.ts

```diff
@@ -47,6 +47,14 @@ export interface MultiInputFieldRefs {
   unstableEndFieldRef?: React.Ref<FieldRef<RangeFieldSection>>;
 }
 
+export interface RangeFieldSeparatorProps {
+  /**
+   * String displayed between the start and the end date.
+   * @default "–"
```

Comment:
In this case, we disallow users to remove the space between dates in case they want/need it. 🤔 
Objectively, it does not seem logical and should not be a problem for most users, but I'm a bit torn on this one and would probably lean 1% more towards going with a less opinionated approach and having the spaces in the `dateSeparator`.
WDYT? Do you see any potential problems with it? 🤔

Reply:
> In this case, we disallow users to remove the space between dates in case they want/need it

Yes, the trade-off is basically do we prefer to make the life of the vast majority of people slightly more complex (having to include the spaces in the separator) to support and edge case?
I think the decision is not super super important because on one hand the edge case seems super weird and rare, but on the other hand the added complexity is also very light.

I don't have a strong opinion here, the change is super easy, I can quickly ask for the team's opinion during the meeting.

Reply:
IMHO there is no right or wrong answer here.
We can keep the existing solution and adjust it in the next major if necessary. 👌